docs/how-to/build-generator-to-file-pipeline.mdx

```diff
@@ -0,0 +1,172 @@
+---
+title: "How to build a generator-to-file pipeline"
+sidebar_position: 0
+---
+
+In this guide, we'll learn how to build a basic Conduit pipeline. It will use
+two built-in connectors:
+
+- A generator source (a source connector that generates random data). Our
+  pipeline will generate imaginary employee data.
+- A logging destination (a destination that simply logs all incoming records).
+
+As you'll see, all it takes to accomplish this is to write
+a [pipeline configuration file](/docs/pipeline-configuration-files/getting-started)!
+
+The steps below will guide you through installing Conduit and gradually building
+the pipeline. If you're in a hurry, you can skip ahead
+and [run the pipeline](#step-6-run-the-pipeline) immediately.
+
+## Step 1: Install Conduit
+
+Download
+the [latest Conduit release](https://github.com/ConduitIO/conduit/releases/latest)
+to a directory of your choice. In this guide, the directory will be
+`~/conduit-playground`.
+
+## Step 2: Create a pipeline configuration file
+
+Pipelines can be created through YAML configuration files, that are, by default, put into a directory called `pipelines`.
+
+With the following command, we'll create the `pipelines` directory and a file that will contain our pipeline's configuration:
+
+```bash
+mkdir pipelines && touch pipelines/generator-to-file.yaml
+```
+The directory layout should be as below:
+
+```console
+~/conduit-playground
+├── conduit
+└── pipelines
+    └── generator-to-file.yaml
+```
+
+
+## Step 3: The pipeline's basics
+
+Next, write the following to `generator-to-file.yaml`:
+
+```yaml
+version: "2.2"
+pipelines:
+    - id: pipeline1
+      # Tells Conduit to run the pipeline automatically
+      status: running
+      name: pipeline1
+      description: A generator-to-file pipeline
+      # The connectors will be configured in the following steps
+      connectors:
+        
+```
+
+## Step 4: Add the generator source
+
+Now we can add a source connector under `connectors` in the configuration file:
+
+```yaml
+connectors:
+  - id: source1
+    type: source
+    plugin: builtin:generator
+    settings:
+      format.type: structured
+      format.options.id: int
+      format.options.name: string
+      recordCount: "1"
```

Comment:
Here I'd actually indent this, since it's nested under the first pipeline. And I suggest using `rate` instead of `recordCount`. In fact, I'd remove the `recordCount` completely from the connector tbh, it's not in the spirit of Conduit, as pipelines should run forever.
```suggestion
    connectors:
      - id: source1
        type: source
        plugin: builtin:generator
        settings:
          format.type: structured
          format.options.id: int
          format.options.name: string
          rate: "1s"
```

Reply:
I've put recordCount because it's just an example and it helps find a few things in logs that I highlighted later on. With more records and moving logs that would be harder to do.

As for the indentation, I didn't `connectors` because it's just a snippet. The focus is on the `connectors` themselves, not the whole pipeline.

Reply:
Well we could set even a lower rate, say `2s` 🤷 As long as it shows what an actual pipeline in Conduit is supposed to be doing, it's not supposed to produce 1 record and stop forever 🙈

Re indenting - how about disabling auto formatting in the editor? The tools should help us, not get in our way.

Reply:
Sure, 2s sounds good to me. :+1: 

As for indenting, I don't think it's about settings, and even if it is, editors are simply set up in different ways. E.g. when you enter a new line, some will go to the beginning of the next line, and some will stay at the same indentation level, and so on. Basically, I don't think that indenting the snippets here will be that much of a help.

Reply:
I pushed a commit with the updated formatting, I pointed it out in another comment ([here](https://github.com/ConduitIO/conduit-site/pull/137#discussion_r1784791551)), it includes just the updated formatting. Maybe we are talking about different things, I felt like that would be faster.

Reply:
No problems, that's good too, I don't think it's a thing that deserves more than a couple of comments anyway.:)